specs/badge.md

```diff
@@ -0,0 +1,80 @@
+# fast-badge
+
+## Overview
+
+*Badge* component is used to highlight an item and attract attention or flag status, such as “New”, “Sale”, or other short phrases.
+
+### Use Cases
+Keep the *badge* label short and succinct &ndash; it should not be a sentence or a long phrase (recommend 2-3 words). Short phrases work well, such as “App of the day,” but keep in mind that they will generally become longer when localized. Keep the text to nouns and adjectives if possible (e.g. “New” or “New today”), so the user won’t think it is an actionable element.
+
+Typical use cases include, but are not limited to, denoting a sale or new item and flagging an item as part of a category
+  
+### Features
+
+A badge should allow the following properties:
+- `fill`, a string that passes the color value for the backplate. Defaults to "transparent".
+- `size`, the size the badge is rendered.
```

Comment:
When thinking about size, how should we quantify that? I'm not sure small and large would be scalable. Could you expand on if these are CSS properties and if so, how they might support the variations in size across design systems?

Reply:
Maybe we just leave out size. We can default to values mapped to our designSystem and app authors can just override if they need to. Or we do Nicks suggestion for `fill`.

`size` property could map to author-defined custom properties.

```
fast-badge {
   --size-small: 12px;
   --size-medium: 16px;
   --size-large: 20px;
}
```
Then used like:

```
<fast-badge size="small" />
<fast-badge fill="large" />
```

Reply:
I think the concept of a map here similar to fill would add value 👍

Reply:
@eljefe223 unresolving this for the time being to bring attention back to the naming / methodology of setting "size". I'm still concerned that small, medium, large will not scale. What happens if we want to or need to add a new size between any of these? Alternatively, what happens if we want a larger or smaller size to be added? Large no longer is large, or we need to add "larger"

Reply:
What do you propose? 1,2,3?

Reply:
@eljefe223,  I think @chrisdholt is proposing (correct me if I'm wrong) using the same implementation that "fill" does for this component, transforming the value of this attribute into CSS Custom Properties used by the component.

Reply:
Yes, I think that by providing a methodology for applying *any* size measurement with preferred semantics allows me to have "one size fits all" badge, or "small, medium, large", or anything.

Reply:
@nicholasrice I'm confused... We talked about this remember? Originally I proposed this, but the issue is that size effects more than one property. `font-size` and `line-height` for the standard badge and `min-width` and `height` for the circular badge. Where `fill` can map to a variable since it only changes one property.

Reply:
I don't think we want to create `attr` for each of those properties.

Reply:
Ahh yes. Perhaps we just make badge support these size classes (instead of a size attribute):

```html
<fast-badge class="small"></fast-badge>
```
```css
:host(.small) { /* small styles */ }
```

That is extensible by authors to define new "sizes" or styles as needed.

Thoughts?

Reply:
I think the semantics of how we are defining size size is a primary problem here. If we define small, medium and large while I can add my own new sizes using classes, they may not be intuitively extensible. For instance, if I want a smaller size, I could use "smaller" as a class name. Similarly if I want something larger than large, that seems reasonable as well. However we have always seen problems when we want something between small and medium or between medium and large. The naming of small, medium, large is all relative to one another. If my design system, or even if our own system calls for a size between any of these, we now need to come up with something less-semantic or intuitive.

Reply:
Do you have a proposal?

Reply:
I just walked through the examples below in prior art/examples and I actually don't see a primary use case for size in the way we are using it outside of the existing FAST implementation. Others do have the ability to have more simple badging with just a dot or indicator, which *could* correlate to size, but it's not their preferred method of identifying that variation. I'm wondering if size common enough to merit first class support over something more common like the dot variation of the others below.

In terms of what it means for size, I would expect that the visual size of the badge would respond based on the content - such that if I applied a larger instance of the type ramp the badge would grow to accommodate it. I do wonder if there is a way to drop "size" and build this in a way that the badge visuals respond well to different text sizes and manage "size" that way.

Reply:
Resolving based on our conversation we had offline. Removing `size` from api